src/library_exceptions.js

```diff
@@ -335,14 +335,14 @@ var LibraryExceptions = {
     var thrown = exceptionLast;
     if (!thrown) {
       // just pass through the null ptr
-      {{{ makeStructuralReturn([0, 0]) }}};
+      return 0;
     }
     var info = new ExceptionInfo(thrown);
     info.set_adjusted_ptr(thrown);
     var thrownType = info.get_type();
     if (!thrownType) {
       // just pass through the thrown ptr
-      {{{ makeStructuralReturn(['thrown', 0]) }}};
+      return 0;
```

Comment:
`thrown` should be returned here, and `setTempRet0(0)` as well (as there may be an old/stale value there).

Reply:
(also in some other places in this file, I think)

Reply:
Hmm.. interesting that the test all pass without that.